src/hooks/routing/clientSideSmartOrderRouter.ts

```diff
@@ -111,13 +111,13 @@ async function getQuote(
   const amount = CurrencyAmount.fromRawAmount(baseCurrency, JSBI.BigInt(amountRaw ?? '1')) // a null amountRaw should initialize the route
   const route = await router.route(amount, quoteCurrency, tradeType, /*swapConfig=*/ undefined, routerConfig)
 
-  if (!amountRaw) return INITIALIZED
-  if (!route) return NO_ROUTE
+  if (!amountRaw) return { state: QuoteState.INITIALIZED }
```

Comment:
when does this case happen? shouldn't this be part of the skipToken case?

Reply:
the slow part of CSSOR is initialization - that first time a token pair is checked. but it's not even getting the quote, it's initializing the data to get the quote.
so as soon as we have a pair, we make a query with a null amount, signifying that it's just an initialization. the client-side code will replace null with 1 so that it fetches the necessary initializing data, and then return QuoteState.INITIALIZED.
then, when we make a real query, it's nearly instant!

Reply:
🤦‍♀️ oy VEY!